### PR TITLE
Test using custom Dataset mapping

### PR DESCRIPTION
**Pull request by uri-granta**

### CODE REVIEW

**Comment by sc336** [comment]
Any chance of reverting or otherwise removing the formatting commits?

**Comment by uri-granta** [comment]
> Any chance of reverting or otherwise removing the formatting commits?

Moved to https://github.com/secondmind-labs/trieste/pull/676

File: tests/unit/acquisition/test_interface.py [hunk 1]
```diff
@@ -144,3 +146,47 @@ def test_single_model_acquisition_function_builder_reprs(
...
+            assert set(datasets) == {OBJECTIVE}
+            assert len(datasets[OBJECTIVE]) == 0
+            assert "FOO" not in datasets
+            assert isinstance(datasets, CustomDatasets)
+            assert datasets.iteration_id == 2
+            return raise_exc
+
+    data = CustomDatasets({OBJECTIVE: empty_dataset([1], [1])}, 2)
+    models = {OBJECTIVE: QuadraticMeanAndRBFKernel()}
+    _CustomData().prepare_acquisition_function(models, data)
```
Comment (sc336) [inline tests/unit/acquisition/test_interface.py]: It's not obvious what is being tested here - any chance of an explanatory comment?

**Review by sc336 — APPROVED** [review]

File: tests/unit/acquisition/test_interface.py (hunk 1 above)
Comment (uri-granta) [reply tests/unit/acquisition/test_interface.py]: Added

**Pull request merged**